docs: Maven - Ensure changelog is fetchable (#16389)
renovatebot/renovate: lib/modules/datasource/maven/readme.md

### Fetching changelogs from API

> In case your repository doesn't hold a `CHANGELOG.md` file and you need to fetch changelogs from an API (just like the one from [Gitlab](https://docs.gitlab.com/ee/api/releases) or [Github](https://docs.github.com/en/rest/releases)), you need to configure `Maven` so that `renovatebot` is able to determine the `sourceUrl` when fetching info about the artifact.

Reviewer: ```suggestion

In case your repository doesn't hold a `CHANGELOG.md` file and you need to fetch changelogs from an API (just like the one from [Gitlab](https://docs.gitlab.com/ee/api/releases) or [Github](https://docs.github.com/en/rest/releases)), you need to configure `Maven` so that `renovatebot` is able to determine the `sourceUrl` when fetching info about the artifact.
```

Add new line.

> In order to do so, you just need to properly configure the [scm section](https://maven.apache.org/scm/git.html) on the `pom.xml` of the given artifact.

Reviewer: ```suggestion
To do so, you need to configure the [scm section](https://maven.apache.org/scm/git.html) on the `pom.xml` of the given artifact.
```

> Example:

Reviewer: ```suggestion
For example:

```

```xml
<scm>
    <url>scm:git:https://github.com/path_to_repository</url>
</scm>
```

> __Note:__ This also works for private repositories. It can leverage on the same `token` that you had to configure in order to be able to reach the private artifactory.

Reviewer: ```suggestion
<!-- prettier-ignore -->
!!! note
    This also works for private repositories. It can leverage the same `token` that you had to configure in order to be able to reach the private artifactory.
```

### Fetching changelogs from API

> In case your repository doesn't hold a `CHANGELOG.md` file and you need to fetch changelogs from an API (just like the one from [Gitlab](https://docs.gitlab.com/ee/api/releases) or [Github](https://docs.github.com/en/rest/releases)), you need to configure `Maven` so that `renovatebot` is able to determine the `sourceUrl` when fetching info about the artifact.

Reviewer: This sentence is too long. It needs a rewrite so it:

- Explains better _why_ one may need to fetch changelogs from a API. I think Renovate can usually find changelogs on its own. So what makes this different?
- Explains if this is aimed at end-users of Renovate, or at maintainers of repositories that Renovate can fetch changelogs from?

Author: Ok. I will rewrite to be more clear.

Is true, Renovate can find changelogs automatically. It may do so from the `CHANGELOG.md` file, but it can also fetch them from the API alternatively. For it to work, it requires the `sourceUrl` to be resolved and that can only happen if it is defined in the artifact's pom. In fact, I'm thinking that may be the case on both situations, so maybe it doesn't make sense to specifically refer the API case.
This is aimed at the person publishing the artifact, so that the end-users can see changelogs on the PRs. In my specific case I am both (publisher and end-user), so I was biased when writing this. I will make that also more clear.

Thanks

Author: Published a new version. Please have a look 🙂

> To do so, you need to configure the [scm section](https://maven.apache.org/scm/git.html) on the `pom.xml` of the given artifact.

Reviewer: It sounds like the goal with this sentence is to explain that you put the path to your repository in the `pom.xml` of the given artifact?

Author: Yes, exactly.

For example:

```xml
<scm>
    <url>scm:git:https://github.com/path_to_repository</url>
</scm>
```

> <!-- prettier-ignore -->

Reviewer: ```suggestion

<!-- prettier-ignore -->
```

Add newline.

!!! note

>     This also works for private repositories. It can leverage the same `token` that you had to configure in order to be able to reach the private artifactory.

Reviewer: ```suggestion
    This also works for private repositories.
    It can leverage the same `token that you had to configure in order to be able to reach the private Artifactory.
```

Artifactory is a brand name, so we should capitalize it. We should also put the next sentence on it's own newline, but with 4 spaces, so it's still part of the admonitions block.
